src/MementoContainer/Analysis/PropertyAnalyzer.cs

```diff
@@ -128,11 +149,31 @@ public IList<IPropertyData> GetProperties(object obj)
         }
 
         /// <summary>
+        /// Tells if a given property has to be registered in the memento in case the class is decorated with MementoClassAttribute
+        /// </summary>
+        /// <param name="kv">Property info and its attributes</param>
+        /// <returns>True if the property must be registered in the memento</returns>
+        protected virtual bool PropertySelectorForMementoClass(KeyValuePair<PropertyInfo, IList<Attribute>> kv)
```

Comment:
Instead of letting users customize the behaviour via inheritance+override, how about via constructor injection?

```csharp
delegate bool PropertyFilter(MementoClassAttribute mementoClassAttr, PropertyInfo pi, IList<Attribute> propertyAttrs);

public class PropertyAnalyzer : IPropertyAnalyzer
{
    public PropertyFilter { get; }

    public PropertyAnalyzer(PropertyFilter filter)
    {
        _filter = filter;
    }

    public PropertyAnalyzer() : this(DefaultFilter)
    {
    }

    private readonly PropertyFilter DefaultFilter = (mementoClassAttr, pi, propertyAttrs) =>
        mementoClassAttr != null
            ? propertyAtrrs.Any(attr => attr is MementoPropertyAttribute) && pi.HasGetAndSet()
            : propertyAtrrs.Any(attr => attr is MementoPropertyAttribute);
}
```

Reply:
I'm more used to work via inheritance than passing certain functions due to our development patterns, but in the end the result is the same, so I have no problem approaching it that way. I'll review the code, make all the changes and push them so you can check them.

Reply:
Thank you! The reason for my suggestion was that composition is usually more flexible and easy to follow than inheritance, as evidenced by the ["composition over inheritance" design principle](https://en.wikipedia.org/wiki/Composition_over_inheritance).